src/client/index.ts

```diff
@@ -98,10 +103,19 @@ export class MonoClient<
     if (isSuccessful) {
       return response;
     }
+
     if (attempt + 1 < maxAttempt && this.shouldRetry(request, response)) {
       return this.requestAttempt({ request, maxAttempt, attempt: attempt + 1 });
     }
-    throw new RequestFail(this.config.type, request, response);
+
+    const error =
+      isSuccessfulResponse instanceof Error
+        ? isSuccessfulResponse
+        : typeof isSuccessfulResponse === 'string'
+        ? new Error(isSuccessfulResponse)
+        : new Error(JSON.stringify(response.body) || 'unknown error');
```

Comment:
This should be `JSON.stringify(response.body ?? 'unknown error')``. Otherwise it will always use JSON.stringify even if response.body is null or undefined

Reply:
if this statement is correct the next line of the test should fail:
https://github.com/comparaonline/mono-client/blob/5a2144f57f73d97ecbd412b58050a5b201328aff/src/client/__tests__/rest.test.ts#L204